[Mobile] - Support background colors for Group block

geriux opened the pull request:
`Gutenberg Mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/2703
`WordPress iOS PR` -> https://github.com/wordpress-mobile/WordPress-iOS/pull/15079
`WordPress Android PR` -> https://github.com/wordpress-mobile/WordPress-Android/pull/13279

Part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/2195

## Description
This PR adds rendering colors support for:
* **Group**: rendering background colors.
* **Quote**: inherits text colors from parent blocks, e.g Group.
* **Paragraph**: rendering text/background colors and adding paddings when a background color is set.

## How has this been tested?

<details><summary>Steps to test: Group/Paragraph</summary>

---
<kbd><img src="https://user-images.githubusercontent.com/4885740/98130885-05cd7280-1ebb-11eb-9ecb-465d5c32d870.png" width="150" /></kbd>

**Steps**:

**From the web editor**

- Create a Group block
- Set a background color and a text color in the Color settings menu
- Add a Paragraph block and write a text
- **Expect** to see the text color of the block inherited from the Group block
- Save the post
- Open the post from the app
- **Expect** to see the correct paddings and colors

**Or use this code**:

```
<!-- wp:group {"backgroundColor":"accent","textColor":"background"} -->
<div class="wp-block-group has-background-color has-accent-background-color has-text-color has-background"><div class="wp-block-group__inner-container"><!-- wp:paragraph -->
<p>Paragraph 1</p>
<!-- /wp:paragraph -->

<!-- wp:group {"style":{"color":{"background":"#0e6056","text":"#02d414"}}} -->
<div class="wp-block-group has-text-color has-background" style="background-color:#0e6056;color:#02d414"><div class="wp-block-group__inner-container"><!-- wp:paragraph -->
<p>Paragraph 2</p>
<!-- /wp:paragraph --></div></div>
<!-- /wp:group -->

<!-- wp:group {"style":{"color":{"background":"#605582","text":"#fbd0ff"}}} -->
<div class="wp-block-group has-text-color has-background" style="background-color:#605582;color:#fbd0ff"><div class="wp-block-group__inner-container"><!-- wp:paragraph -->
<p>Paragraph 3</p>
<!-- /wp:paragraph -->

<!-- wp:group {"style":{"color":{"background":"#ee2865","text":"#ecd0f2"}}} -->
<div class="wp-block-group has-text-color has-background" style="background-color:#ee2865;color:#ecd0f2"><div class="wp-block-group__inner-container"><!-- wp:paragraph -->
<p>Paragraph 4</p>
<!-- /wp:paragraph --></div></div>
<!-- /wp:group --></div></div>
<!-- /wp:group -->

<!-- wp:paragraph -->
<p>Paragraph 5</p>
<!-- /wp:paragraph --></div></div>
<!-- /wp:group -->
```

</details>

<details><summary>Steps to test: Quote</summary>

---
<kbd><img src="https://user-images.githubusercontent.com/4885740/98134876-7ffff600-1ebf-11eb-9e73-000ff8100880.png" width="150" /></kbd>

**Steps**:

**From the web editor** 

**Note**: You won't see the colors applied to the Quote block in the web editor

- Create a Group block
- Set a text color in the Color settings menu
- Add a Quote block
- Set some text in the block
- Save the post
- Open the post from the app
- **Expect** to see the Quote block inheriting its parent's text color set in the web editor.

**Or use this code**:

```
<!-- wp:group {"backgroundColor":"background","style":{"color":{"text":"#4d4313"}}} -->
<div class="wp-block-group has-background-background-color has-text-color has-background" style="color:#4d4313"><div class="wp-block-group__inner-container"><!-- wp:quote -->
<blockquote class="wp-block-quote"><p>This is my cool quote</p><cite>Citation here</cite></blockquote>
<!-- /wp:quote --></div></div>
<!-- /wp:group -->
```
</details>


## Screenshots <!-- if applicable -->
iOS | Web editor
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/95600575-a793a800-0a52-11eb-8d37-3555d8ead6a6.png" width="250" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/95600576-a8c4d500-0a52-11eb-8913-4a1e0aeccbc5.png" width="250" /></kbd> 

Template Example | Template Example
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/95600941-1c66e200-0a53-11eb-9970-34f0a4d561fd.png" width="250" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/95600637-c003c280-0a52-11eb-8472-fb6b470f2248.png" width="250" /></kbd> 


## Types of changes
New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

Commit:
Mobile - Support background/foreground colors for Group block

kyleaparker commented:
Nice, this is working well for me!

iamthomasbishop commented:
@geriux This is looking solid, nice work! I found one issue, but I'm not sure if it's related to the changes you've made — I'm guessing it's specific to recent work done on the BottomSheet component.

The issue is on block settings sheets; when you open them, they get stuck at ~40-60px height. Here's a screenshot:

![image](https://user-images.githubusercontent.com/1200432/95762986-739bcb00-0c7c-11eb-9cb7-e250d09e3c15.jpeg)

And fwiw, I'm only seeing it happen on these blocks (and not only in the context of Group block):
 
- Heading
- Group
- List
- Media & Text

kyleaparker commented:
I just noticed after making some edits on a page with this build, it starts to give me these errors on the desktop editor: 
<img width="981" alt="Edit Page ‹ Overton — WordPress com 2020-10-13 18-53-31" src="https://user-images.githubusercontent.com/712873/95820987-9cf44e00-0d85-11eb-8513-b54bb5f4dd27.png">

The page was working fine on the web immediately before I made changes in the mobile editor.

Commit:
Merge branch 'master' into rnmobile/feature/group-background-color

geriux commented:
> I found one issue, but I'm not sure if it's related to the changes you've made — I'm guessing it's specific to recent work done on the BottomSheet component.

> The issue is on block settings sheets; when you open them, they get stuck at ~40-60px height. Here's a screenshot:

Oops, forgot to disable that 😅. The issue you're seeing is that it's trying to show the `Color Settings` since the block has colors set but this is not available on mobile yet.

Thanks for testing @iamthomasbishop!

geriux commented:
> I just noticed after making some edits on a page with this build, it starts to give me these errors on the desktop editor:

There's an ongoing issue with the Columns block and it's included in the build I created so sorry for that, hope you were able to recover your changes 🤞 

Once this issue is solved I'll do more testing to double check is not giving any issues between the app and the web editor.

Thanks for testing @kyleaparker!

Commit:
Mobile - Enable ColorEdit

geriux commented:
@iamthomasbishop The issue is now solved, I added the Color picker to the blocks that support text/background colors:

<kbd><img src="https://user-images.githubusercontent.com/4885740/95881408-d6728c80-0d78-11eb-9507-faf2bd47fc2a.gif" width="250" /></kbd>

The build now includes this just in case you wanted to try it out =)

iamthomasbishop commented:
This looks great! Not sure why, but I didn't realize we were also adding color support for non-Group blocks — so exciting! Are we planning on adding some padding to any block with `.has-background`? For example: 

<a href="https://user-images.githubusercontent.com/1200432/95909564-55a99580-0d64-11eb-8129-d5f084259633.jpeg"><img src="https://user-images.githubusercontent.com/1200432/95909564-55a99580-0d64-11eb-8129-d5f084259633.jpeg" width="375"></a>

As a side note, playing around with this makes me think we should add an explicit default color button on the color sheets so the user can "reset" back to the default. Perhaps at the bottom right of the sheet like so? // cc @lukewalczak 

<a href="https://user-images.githubusercontent.com/1200432/95909817-bc2eb380-0d64-11eb-88ab-4df6642cb9d9.jpeg"><img src="https://user-images.githubusercontent.com/1200432/95909817-bc2eb380-0d64-11eb-88ab-4df6642cb9d9.jpeg" width="375"></a>

Commit:
Mobile - Add paddings to text blocks with background color

Commit:
Mobile - Add clear button to palette color screen

Commit:
Merge branch 'master' into rnmobile/feature/group-background-color

geriux commented:
Hey @iamthomasbishop 👋 

> This looks great! Not sure why, but I didn't realize we were also adding color support for non-Group blocks — so exciting! Are we planning on adding some padding to any block with `.has-background`? For example:

Well, when I activated it for Groups I saw those were working too so I didn't limit this feature for Groups just yet.

I just added paddings for those as well, can you check and let me know if it works as expected?

> As a side note, playing around with this makes me think we should add an explicit default color button on the color sheets so the user can "reset" back to the default. Perhaps at the bottom right of the sheet like so?

Yeah I saw we might need that when I was testing and I couldn't remove any colors =D what do you think?:

<kbd><img src="https://user-images.githubusercontent.com/4885740/96013598-beb50a00-0e45-11eb-9051-57f39cdb276d.gif" width=250 /></kbd>

Thanks for the feedback and for testing! I updated the test build with the latest changes.

Commit:
Mobile - Global styles context utils and tests

iamthomasbishop commented:
@geriux Thanks for the updated build and the quick turn-around!

> Yeah I saw we might need that when I was testing and I couldn't remove any colors =D what do you think?:

Positioning looks good, but I think "Reset" or "Use default" for the label might be more appropriate. I also think we might want to keep the user on that same sheet after they've pressed the clear button, although that might just be a personal preference.

> I just added paddings for those as well, can you check and let me know if it works as expected?

Paddings work as I would expect. Nice work!

------------

One thing I wanted to raise — I am concerned about the side-effects WRT light vs. dark modes. For example, let's say I have dark mode enabled. If I change the background of a Group to dark gray, it'll look fine in the editor in dark mode because the base text color is white. However, if I preview it, I will see that the output is black text because the theme's base text color is black. Here's what I mean:

| **Editor** | **Preview** |
| --------------- | --------------- |
| <a href="https://cldup.com/h0rhwGbh6W.jpeg"><img src="https://cldup.com/h0rhwGbh6W.jpeg" width="375"></a> | <a href="https://cldup.com/xPDUbnAwjw.jpeg"><img src="https://cldup.com/xPDUbnAwjw.jpeg" width="375"></a>


I think if we had a mechanism to calculate color contrast and adjust the color of an element based on the ratio between it and another element's color (i.e. text color vs background color), we would be in great shape. But as it stands, we'll be stuck in the middle — and I'm not sure if it's a good idea to ship in this state. 

Another thing that might help is if we can read and apply the theme's base color values (likely with global styles?), but until then we'll need to figure out a way to support both light and dark modes dynamically. I believe the web has a mechanism for auto-converting the text color based on the color behind it, but they also have a color contrast mechanism that alerts the user when the color contrast ratio between two colors isn't sufficient — perhaps we can utilize these in some way?

kyleaparker commented:
I've been trying to edit some of the homepage layout designs on this build to see how they look with the new colour settings and the page seems to freeze each time, I have been going back and forth between the mobile and web editor so it may be related to that issue? The app is also crashing when I try to add the spacer block to any post/page. I can send more details about the content on the homepages if you need :)

Commit:
Merge branch 'master' into rnmobile/feature/group-background-color

geriux commented:
Hey @iamthomasbishop thanks for giving it another test 👋 

> Positioning looks good, but I think "Reset" or "Use default" for the label might be more appropriate. I also think we might want to keep the user on that same sheet after they've pressed the clear button, although that might just be a personal preference.

Sounds good! I'll change it to "Reset" and keep the user on the same sheet.

> Paddings work as I would expect. Nice work!

Yay!

> I think if we had a mechanism to calculate color contrast and adjust the color of an element based on the ratio between it and another element's color (i.e. text color vs background color), we would be in great shape. But as it stands, we'll be stuck in the middle — and I'm not sure if it's a good idea to ship in this state.

> Another thing that might help is if we can read and apply the theme's base color values (likely with global styles?), but until then we'll need to figure out a way to support both light and dark modes dynamically. I believe the web has a mechanism for auto-converting the text color based on the color behind it, but they also have a color contrast mechanism that alerts the user when the color contrast ratio between two colors isn't sufficient — perhaps we can utilize these in some way?

The thing is we are still missing information from the theme like the default text color, or background. What we currently have is all of the colors the theme uses but even with that, we don't really know what color to pick. Of course, we can compare the background color with the colors from the theme and pick the light color if the background is dark but it might have different results from web.

What web does is just using the CSS classes from the theme and applying the text colors for each background. As of today, we don't have a way to get that.

Regarding the color contrast mechanism, we could use that but we need both background and text color to compare. If the user doesn't select a text color we wouldn't be able to know if the background color is right for the theme text color.

If we don't want to ship this like this, we could disable the color picker unless a block already has colors set. So for layouts, they could tweak them. But if they create a new block there won't be an option to customize colors.

To be honest (Light mode / Dark mode) will always be an issue with blocks with custom colors =(

geriux commented:
Hey @kyleaparker 👋 

> I've been trying to edit some of the homepage layout designs on this build to see how they look with the new colour settings and the page seems to freeze each time, I have been going back and forth between the mobile and web editor so it may be related to that issue? The app is also crashing when I try to add the spacer block to any post/page. I can send more details about the content on the homepages if you need :)

So sorry for that, we had recent issues that broke some blocks. I'll update the build with the fixes.

Thanks for testing!

Commit:
Mobile - Palette screen - Change text and keep the same sheet when resetting a color

iamthomasbishop commented:
@geriux thanks for the context, really great info.

> What we currently have is all of the colors the theme uses but even with that, we don't really know what color to pick

That's a good point. I hope in the future (via global styles, perhaps) we will be able to receive some registered color properties straight from the theme, but until then it seems like we're stuck.

> If we don't want to ship this like this, we could disable the color picker unless a block already has colors set. So for layouts, they could tweak them. But if they create a new block there won't be an option to customize colors.

I think that's an option (albeit a "nuclear" one), but let's try to think about some alternatives before going there.

> To be honest (Light mode / Dark mode) will always be an issue with blocks with custom colors =( 

We can move this discussion to another ticket if necessary, especially if we decide on a more robust approach, but I wanted to note a couple of ideas that might be worth considering in case it sparks ideas for a quick solution. Some of these are straight forward, and others conceptual departures from the current approach, but we need to come up with something until we have global styles or similar access to the theme's various properties.

* **Option A**: Alert the user (in the color settings bottom sheet or via a dialog/alert) that colors may not match up, especially when they have the default value applied, so they're aware of a potential mismatch. 
  * This is probably the most straight forward in terms of implementation, but opens some potentially confusion, especially if the user doesn't read the messaging (let's be honest, most users don't read this type of messaging 🤷‍♂️).

* **Option B**: Upon opening the editor, present the user with an explicit choice between light mode and dark mode ("Does your site use a light or dark theme?"). 
  * Based on that selection, we display the canvas as such — note: the editor chrome would follow the system appearance setting — "only" the canvas contents (blocks and their contents) would be affected by this. We could also allow them to override this setting in the editor-level `•••` menu, but we'd likely want to warn about the potential side-effects.  
  * **Question on this point**: I know the WP.com [theme directory](https://wordpress.com/themes/filter/style:dark) has a filter for style, so I wonder if we're able to default to light/dark based on a property/tag attached to the theme? 

* **Option C**: Force/default to light mode. 
  * This is sorta a nuclear option, but would simplify things a lot for most users. We could keep dark mode as an option, but it would need to be turned on manually. The problem is that it assumes that most users have light-background themes — an assumption that AFAIK we can't back up quantitatively. Also, even if _most_ themes have light backgrounds, it is not always a binary thing — any give page or post can have sections with varying background/text colors. There are probably other blindspots to this approach, but it came up in a team discussion yesterday. // cc @mattmiklic @kyleaparker 

* **Your ideas here**: If you have other ideas (yes, you!), please feel free to share! 😄 

// cc @maxme any ideas re: ⬆️ ?

Commit:
Merge branch 'master' into rnmobile/feature/group-background-color

Commit:
Mobile - Quote block - Support to inherit colors from parent block

geriux commented:
Hey @iamthomasbishop 👋 thanks for the feedback and different options!

> * **Option A**: Alert the user (in the color settings bottom sheet or via a dialog/alert) that colors may not match up, especially when they have the default value applied, so they're aware of a potential mismatch.
>   
>   * This is probably the most straight forward in terms of implementation, but opens some potentially confusion, especially if the user doesn't read the messaging (let's be honest, most users don't read this type of messaging 🤷‍♂️).

I agree this one would be the most straight forward option, having a text as we do for other cases within the BottomSheet, e.g:

<kbd><img src="https://user-images.githubusercontent.com/4885740/98010354-dd7c4000-1df6-11eb-988d-9c3c1bcf6319.png" width=100 /></kbd>

Although I'd try to avoid dialogs but that's a personal preference, I think most users won't even read it just as with the message.

For this task though, I think I went a bit further enabling the color settings when we still haven't completed the global styles work. Since the goal for this PR is to just render the colors of the layouts maybe the easiest solution (using what we have now) is just disabling the option to customize colors (For Group, Paragraph, etc) until we could offer a better way. What do you think?

Depending on what we decide we should definitely move this conversation and your suggested options to a new ticket to keep discussing as we continue the work to support Global Styles.

Commit:
Merge branch 'master' into rnmobile/feature/group-background-color

geriux commented:
Hey @kyleaparker 👋 I've updated the [build](https://github.com/wordpress-mobile/WordPress-iOS/pull/15079#issuecomment-706263686) with the Quote block color changes and all the fixes for the previous issues you had. Would you mind giving it another quick test to make sure all is good? Thank you!

kyleaparker commented:
Thanks @geriux! I've checked all the home pages and they all look good to me. 

The only issue was when I first loaded some of the homepages, the text colors didn't render, but when I closed and reloaded the page they rendered correctly. I assume it was a caching issue, but thought I should mention it.

kyleaparker commented:
>Depending on what we decide we should definitely move this conversation and your suggested options to a new ticket to keep discussing as we continue the work to support Global Styles.

Agree we should probably open a separate issue to discuss these options, but I just wanted to add my opinion here. 

>Option C: Force/default to light mode.

>This is sorta a nuclear option, but would simplify things a lot for most users. We could keep dark mode as an option, but it would need to be turned on manually. The problem is that it assumes that most users have light-background themes — an assumption that AFAIK we can't back up quantitatively. Also, even if most themes have light backgrounds, it is not always a binary thing — any give page or post can have sections with varying background/text colors. There are probably other blindspots to this approach, but it came up in a team discussion yesterday. // cc @mattmiklic @kyleaparker

This option would definitely be my preference. It makes sense that the top bar and any settings, bottom sheets etc should render in dark mode, but the site itself should not. The visual design in the editor should match what's on the site, and this seems to be what most users **expect**. Dark mode confuses this, especially if we want to move closer to 1:1 parody between the editor and the site.

geriux commented:
Thanks for testing them again @kyleaparker! Glad to know they're working now =)

> The only issue was when I first loaded some of the homepages, the text colors didn't render, but when I closed and reloaded the page they rendered correctly. I assume it was a caching issue, but thought I should mention it.

I'll investigate this issue that was reported [here](https://github.com/wordpress-mobile/gutenberg-mobile/issues/2637) separately from this PR.

geriux commented:
Hey @dratwas 👋 I'm assigning you as a reviewer for this PR since you worked on the first implementation of this =)

One thing I want to mention is that we still haven't decided if we'll keep the current PR implementation that shows the color picker to customize blocks like Group, Paragraph and what approach we'll take to let the users know that some colors might be different depending on the default colors they have selected in their themes or simply just not allowing them to customize them from the mobile editor yet.

The rest of the implementation is mainly rendering colors/adding paddings to improve the user experience while rendering page layouts on the mobile editor.

Let me know if you have any doubts, thanks!

dratwas commented on packages/rich-text/src/component/index.native.js:
```diff
@@ -827,7 +827,15 @@ export class RichText extends Component {
 		return (
-			<View>
+			<View
+				style={ [
```
I think you can remove the array from here since you add only one object inside :)

geriux replied:
Sounds good! Updated 👍

dratwas commented on packages/primitives/src/block-quotation/index.native.js:
```diff
@@ -13,20 +13,31 @@ import { withPreferredColorScheme } from '@wordpress/compose';
-				style: styles.wpBlockQuoteCitation,
+				style: {
+					...styles.wpBlockQuoteCitation,
+					...( style?.color ? { color: style.color } : {} ),
```
WDYT about moving that check out of map function? I think it would be more performant to do something like:

```js
const colorStyle = style?.color ? { color: style.color } : {}
```

and here:

```js
style: {
	...styles.wpBlockQuoteCitation,
	...colorStyle,
}
```

dratwas commented on packages/primitives/src/block-quotation/index.native.js:
```diff
@@ -13,20 +13,31 @@ import { withPreferredColorScheme } from '@wordpress/compose';
-			return cloneElement( child, { tagsToEliminate: [ 'div' ] } );
+			return cloneElement( child, {
+				tagsToEliminate: [ 'div' ],
+				...( style?.color ? { style: { color: style.color } } : {} ),
```
Same as above, we could use the created const instead of doing that check here :)

dratwas commented on packages/block-editor/src/components/block-list/block.native.js:
```diff
@@ -23,6 +28,66 @@ import BlockEdit from '../block-edit';
+				return (
+					<GlobalStylesContext.Provider value={ mergedStyle }>
+						<BlockEdit
+							name={ name }
+							isSelected={ isSelected }
+							attributes={ attributes }
+							setAttributes={ onChange }
+							onFocus={ onBlockFocus }
+							onReplace={ onReplace }
+							insertBlocksAfter={ insertBlocksAfter }
+							mergeBlocks={ mergeBlocks }
+							onCaretVerticalPositionChange={
+								onCaretVerticalPositionChange
+							}
+							// Block level styles
+							wrapperProps={ wrapperProps }
+							// inherited styles merged with block level styles
+							mergedStyle={ mergedStyle }
+							clientId={ clientId }
+							parentWidth={ parentWidth }
+							contentStyle={ contentStyle }
+							onDeleteBlock={ onDeleteBlock }
```
Seems like all of these props have the same name as destructured ones. WDYT about something like:

```js
function BlockForType( {
  attributes,
  getBlockWidth,
  wrapperProps,
  ...restProps
} )
```
```js
<BlockEdit
  {...restProps}
  // Block level styles
  wrapperProps={ wrapperProps }
  // inherited styles merged with block level styles
  mergedStyle={ mergedStyle }
/>
```

geriux replied:
Although this would be good because it'll remove some lines, sometimes I feel like when doing this it's hard to track props when checking/refactoring code or might cause unexpected results. Still, I tried your suggestion and I started to get issues with some props and its references... 😅

dratwas replied:
Yeah, you're right. Since we don't have typing it would be good to leave it as it is for future refactors :)

Commit:
Mobile - Block quotation and Rich text styles

Commit:
Merge branch 'master' into rnmobile/feature/group-background-color

Commit:
Mobile - Group block - Remove extra bottom padding when a background color is set

geriux commented:
Hey @iamthomasbishop 👋 I created a new build with the fix for the extra bottom padding in Group blocks with a background color. Let me know if you find any other issues. Thanks for testing!

iamthomasbishop commented:
Thanks @geriux! That padding issue looks to be fixed! 👍

Commit:
Mobile - Color settings - Remove clear button, to be added in another PR

Commit:
Mobile - Disable ColorEdit for blocks that have color support

Commit:
Merge branch 'master' into rnmobile/feature/group-background-color

Commit:
Merge branch 'master' into rnmobile/feature/group-background-color

geriux commented:
Hey @dratwas 👋 I've updated the code to disable the Color settings for these blocks, so this PR will only focus only on rendering the colors. Would you mind giving it another review when you can? I also created new builds to test.

Thank you!

dratwas commented on packages/block-library/src/group/edit.native.js:
```diff
@@ -64,6 +68,19 @@ function GroupEdit( {
+				mergedStyle,
+				isSelected &&
+					hasInnerBlocks &&
+					mergedStyle?.backgroundColor && {
+						paddingBottom:
+							styles.hasBackgroundAppender.paddingBottom,
+					},
```
It could be moved to the scss file

```suggestion
					mergedStyle?.backgroundColor && styles.hasBackgroundAppender,
```

dratwas commented on packages/block-library/src/group/edit.native.js:
```diff
@@ -64,6 +68,19 @@ function GroupEdit( {
+					isInnerBlockSelected &&
+					isLastInnerBlock &&
+					mergedStyle?.backgroundColor && {
+						paddingBottom: 0,
```
Same here :) It could be moved to the scss file

dratwas commented on packages/block-library/src/group/edit.native.js:
```diff
@@ -72,14 +89,34 @@ function GroupEdit( {
+		const hasInnerBlocks = !! ( block && block.innerBlocks.length );
+		const isInnerBlockSelected =
+			hasInnerBlocks && hasSelectedInnerBlock( clientId );
+		let isLastInnerBlock = hasInnerBlocks;
```
I'm a bit confused by this name. Does that determine if the last inner block is selected? If yes, then shouldn't it be `isLastInnerBlockSelected`?

Hmmm, what in the case when the block has inner blocks but none of them is selected? The `isLastInner` would be `true`. Is that correct?

geriux replied:
Thanks for the suggestions, I went ahead and changed the name and changed the default value to `false`

dratwas commented on packages/block-library/src/quote/edit.js:
```diff
@@ -24,12 +24,14 @@ export default function QuoteEdit( {
 		className: classnames( className, {
 			[ `has-text-align-${ align }` ]: align,
 		} ),
+		style: mergedStyle,
```
I'm just wondering if that doesn't break anything on the web. I believe the `mergedStyle` is `undefined` for the web but it would be good to confirm that:)

geriux replied:
Thanks for double for checking! I tested this change on the web editor and it works without any issues, we already have this use case in the [Heading](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/heading/edit.js#L37) block.

dratwas commented on packages/components/src/mobile/global-styles-context/utils.native.js:
```diff
@@ -0,0 +1,67 @@
+		( wrapperPropsStyle?.backgroundColor ||
+			blockStyleAttributes?.backgroundColor )
+	) {
+		blockPaddings.padding = PADDING;
```
I think that you could return here

Commit:
Mobile - Group - Background color fixes

dratwas reviewed (approved):
Tested it on iOS and Android and everything works well. Great job @geriux 🎉

Commit:
Mobile - Update changelog
